app/services/course_search.rb

```diff
@@ -13,6 +13,7 @@ def call(**args)
   def call
     scope = Course
       .findable
+      .published
```

Comment:
```ruby
[13] pry(main)> cfpwrc = Course.findable .published.with_recruitment_cycle("2020").size
=> {[42513, 12928759]=>1}
[14] pry(main)> cfpwrc = Course.findable .published.size
=> {[42513, 12928759]=>1}
[15] pry(main)> cfpwrc = Course.findable.size
=> 44968
[16] pry(main)> cfpwrc = Course.findable.with_recruitment_cycle("2020").size
=> 23810
[17] pry(main)> cfpwrc = Course.published.size
=> {[42513, 12928759]=>1}
[18] pry(main)> cfpwrc = Course.all.size
=> 37202
[19] pry(main)> cfpwrc = Course.findable.distinct.size
=> 25003
```

:thinking:

Reply:
Do some of those look wrong?

Reply:
```
> Course.findable .published.with_recruitment_cycle("2020").length
=> 1
> Course.findable.published.with_recruitment_cycle("2020").length
=> 1
```

Why only one course returned by those scopes added together?

Reply:
root cause
`published`

Reply:
This is why:

```
> Course.published.length
=> 1
```

Although I'm still not clear on why yet.

```
SELECT "course".* FROM "course"
       INNER JOIN "course_enrichment" ON "course_enrichment"."course_id" = "course"."id"
       WHERE (course_enrichment.created_at = (
              SELECT MAX(course_enrichment.created_at)
                     FROM course_enrichment))
              AND "course_enrichment"."status" = 1
       GROUP BY course_enrichment.course_id, course_enrichment.id
```

Reply:
Actually, this looks suspicious:

```
course_enrichment.created_at = (
  SELECT MAX(course_enrichment.created_at) FROM course_enrichment)
)
```

Reply:
Tricky one. I think it's something up with the grouping. I've been struggling to get it to work and be able to chain the scopes. The grouping on `course_enrichment` should be in the subquery but it's getting added at the end when the scopes are chained. I'll try some other things.